pipeline/subtraction.py

```diff
@@ -116,8 +130,57 @@ def _subtract_zogy(self, new_image, ref_image):
                 The ZOGY score image (the matched-filter result)
             zogy_psf: np.ndarray
                 The ZOGY PSF image (the matched-filter PSF)
+            zogy_alpha: np.ndarray
+                The ZOGY alpha image (the PSF flux image)
+            zogy_alpha_err: np.ndarray
+                The ZOGY alpha error image (the PSF flux error image)
         """
-        raise NotImplementedError('Not implemented ZOGY subtraction yet')
+        new_image_data = new_image.data
+        ref_image_data = ref_image.data
+        new_image_psf = new_image.psf.get_clip()
+        ref_image_psf = ref_image.psf.get_clip()
+        new_image_noise = new_image.bkg_rms_estimate  # TOOD: improve this by using a Background object?
+        ref_image_noise = 1.0  # proper coaddition images have noise=1.0 by construction
+        new_image_flux_zp = 10 ** (0.4 * new_image.zp.zp)
+        ref_image_flux_zp = 10 ** (0.4 * ref_image.zp.zp)
+        # TODO: consider adding an estimate for the astrometric uncertainty dx, dy
+
+        # do some additional processing of the new image
+        mu, sigma = sigma_clipping(new_image_data)
+        new_image_data = (new_image_data - mu) / sigma  # TODO: skip this if we already background subtracted
```

Comment:
Also TODO would be a way to specify different background subtraction methods.  (Though perhaps in that case you should just background subtract before throwing it into this function.  We'd need to document that clearly though, indicating which background subtraction method it will use if it doesn't get background-subtracted images.)

(I'll have to look at sigma clipping; it might be about the same thing as the sextractor background algorithm.  But, there's also things like the box size for smoothing when using a variable background.  And, for some fields, we might want to use a more complicated background subtraction method.)

Reply:
This is here only as a placeholder. Once we get organized on the background subtraction it will no longer be necessary to do any of this at the subtraction point. You'd just be guaranteed to be able to produce a background subtracted image on the fly because you'd have a Background object, the same way you have a PSF object.